flink-kubernetes/src/test/java/org/apache/flink/kubernetes/KubernetesResourceManagerTest.java

```diff
@@ -0,0 +1,366 @@
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.Callable;
+import java.util.concurrent.CompletableFuture;
+import java.util.stream.Collectors;
+
+import static junit.framework.TestCase.assertEquals;
+import static org.hamcrest.Matchers.instanceOf;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+/**
+ * Test cases for {@link KubernetesResourceManager}.
+ */
+public class KubernetesResourceManagerTest extends KubernetesTestBase {
+
+	private static final Time TIMEOUT = Time.seconds(10L);
+
+	private TestingFatalErrorHandler testingFatalErrorHandler;
+
+	private final String jobManagerHost = "jm-host1";
+
+	private Configuration flinkConfig;
+
+	@Before
+	public void setup() {
+		testingFatalErrorHandler = new TestingFatalErrorHandler();
+		flinkConfig = new Configuration(FLINK_CONFIG);
+		flinkConfig.setString(TaskManagerOptions.TOTAL_PROCESS_MEMORY, "1024m");
+	}
+
+	@After
+	public void teardown() throws Exception {
+		if (testingFatalErrorHandler != null) {
+			testingFatalErrorHandler.rethrowError();
+		}
+	}
+
+	class TestingKubernetesResourceManager extends KubernetesResourceManager {
+
+		private SlotManager slotManager;
+
+		public TestingKubernetesResourceManager(
+				RpcService rpcService,
+				String resourceManagerEndpointId,
+				ResourceID resourceId,
+				Configuration flinkConfig,
+				HighAvailabilityServices highAvailabilityServices,
+				HeartbeatServices heartbeatServices,
+				SlotManager slotManager,
+				JobLeaderIdService jobLeaderIdService,
+				ClusterInformation clusterInformation,
+				FatalErrorHandler fatalErrorHandler,
+				ResourceManagerMetricGroup resourceManagerMetricGroup) {
+			super(
+				rpcService,
+				resourceManagerEndpointId,
+				resourceId,
+				flinkConfig,
+				highAvailabilityServices,
+				heartbeatServices,
+				slotManager,
+				jobLeaderIdService,
+				clusterInformation,
+				fatalErrorHandler,
+				resourceManagerMetricGroup
+			);
+			this.slotManager = slotManager;
+		}
+
+		@Override
+		protected FlinkKubeClient createFlinkKubeClient() {
+			return getFabric8FlinkKubeClient();
+		}
+
+		<T> CompletableFuture<T> runInMainThread(Callable<T> callable) {
+			return callAsync(callable, TIMEOUT);
+		}
+
+		@Override
+		protected void runAsync(Runnable runnable) {
+			runnable.run();
+		}
+
+		MainThreadExecutor getMainThreadExecutorForTesting() {
+			return super.getMainThreadExecutor();
+		}
+
+		SlotManager getSlotManager() {
+			return this.slotManager;
+		}
+	}
+
+	@Test
+	public void testStartAndStopWorker() throws Exception {
+
+		final TestingKubernetesResourceManager resourceManager = this.createAndStartResourceManager(flinkConfig);
+		registerSlotRequest(resourceManager);
```

Comment:
Make creation of `resourceManager` and `registerSlotRequest` inside `setup()`, then all the tests can avoid this process.

Reply:
I will move creation of `resourceManager` to `setup` and keep `registerSlotRequest`. Because we do not `registerSlotRequest` every time and should be called sometimes latter.